posix/netdb.c

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/netdb.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/corelibc/sys/socket.h>
+#include <openenclave/internal/posix/raise.h>
+#include <openenclave/internal/posix/resolver.h>
+#include <openenclave/internal/thread.h>
+#include <openenclave/internal/trace.h>
+
+static size_t _resolver_table_len = 3;
+static oe_resolver_t* _resolver_table[3] = {0}; // At most 3
+static oe_spinlock_t _lock = OE_SPINLOCK_INITIALIZER;
+
+/* Called by the public oe_load_module_host_resolver() function. */
+int oe_register_resolver(int resolver_priority, oe_resolver_t* presolver)
```

Comment:
>resolver_priority [](start = 29, length = 17)

For my understanding, how is this used since `oe_load_module_host_resolver()` is not part of this PR?

- The priority suggests it should behave like registering handlers, though the interaction model for that tends to be a list that you either add at the top or bottom of. Is this only used internally so we always control the ordering and number of resolvers?

- Does the limit on number of resolvers map to a fixed number of priorities (low/med/high?) or known resolver types in resolver.h?

- There's no collision checking for registration of a resolver into the table. Is that expected? If so, shouldn't the previous resolver at least be freed/released?

Reply:
@yakman2020 Bruce: can you take this one?

Reply:
I removed the ability to register multiple resolvers since only a single resolver is supported today. We can revisit this down the road if there is ever more than on resolver.